src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -388,11 +388,22 @@ boolean refersToImpl(T obj) {
     private native boolean refersTo0(Object o);
 
     /**
-     * Clears this reference object.  Invoking this method will not cause this
-     * object to be enqueued.
+     * Clears this reference object. Invoking this method does not enqueue this
+     * object, and the garbage collector will no longer enqueue this object once
+     * the referent reaches the designated reachability level.
+     * <p>
+     * There is a potential race condition with the garbage collector. When this
+     * method is called, the garbage collector may already be in the process of
+     * (or already completed) clearing and/or enqueueing this reference.
+     *
+     * Avoid this race by ensuring the referent remains strongly-reachable until
```

Comment:
`s/strongly-reachable/strongly reachable/`  no dash as in other occurrences.

Reply:
Changed in Cleaner, too